pkg/network/setup/netstat.go

```diff
@@ -168,7 +168,14 @@ func ifacesStatusFromGuestAgent(vmiIfacesStatus []v1.VirtualMachineInstanceNetwo
 	for _, guestAgentInterface := range guestAgentInterfaces {
 		if vmiIfaceStatus := netvmispec.LookupInterfaceStatusByMac(vmiIfacesStatus, guestAgentInterface.Mac); vmiIfaceStatus != nil {
 			updateVMIIfaceStatusWithGuestAgentData(vmiIfaceStatus, guestAgentInterface, vmiInterfacesSpecByName)
-			vmiIfaceStatus.InfoSource = netvmispec.InfoSourceDomainAndGA
+			// For backward compatability with older virt-launchers, apply this logic:
+			// - When the domain status `InterfaceName` field is set, the data originates from the guest-agent.
+			//   This is true for old virt-launchers and new ones alike.
+			// - When the domain status `InterfaceName` field is not set (empty), the data originates from a merge
+			//   at an old virt-launcher and comes from the domain spec. In such cases, no action is to be taken.
+			if guestAgentInterface.InterfaceName != "" {
+				vmiIfaceStatus.InfoSource = netvmispec.InfoSourceDomainAndGA
```

Comment:
I think `updateVMIIfaceStatusWithGuestAgentData` should be moved here as well.

Reply:
That function updates the VMI status with the GA ifaceName and IP/s. Why would this be wrong when treating old vmi-launchers?
For data that originated from GA report, it will be correct.
For data that originated from the domain-only (in the merge), it will also be correct (as this data is not available from the domain).

If we need to compliment the data later on, it will be done later from reading from the podCache.

Reply:
I understand that currently it works as is. But `updateVMIIfaceStatusWithGuestAgentData` is designed to deal with GA data only. For example, it can assume `InterfaceName` is always there. The fact that currently it works on a "fake" GA data doesn't mean it will be the case in the future and I prefer not to maintain this.

But it is not a blocker for me, so if I didn't convince you, you can keep it as is.

Reply:
What I am trying to say is that it works fine for both new and old launchers, just because the data there can only come from the GA. If it did not come from it, it will be empty and it is still correct (because no one else will fill it up until this step).

But when you pointed this function, I realized it can be simplified and remove a condition.
I am pushing a change to simplify it even further.